tests/Integration/Behaviour/Features/Scenario/Product/update_prices.feature

```diff
@@ -0,0 +1,141 @@
+        | ecotax             | 0               |
+        | tax rules group    | US-AL Rate (4%) |
+        | on_sale            | true            |
+        | wholesale_price    | 70              |
+#        #todo: rounding issue. #19620
+#        #| unit_price         | 900         |
+        | unity              | bag of ten      |
+        | unit_price_ratio   | 0.222222        |
+      When I update product "product1" prices with following information:
+        | ecotax              | 5.5            |
+        | on_sale             | false          |
+      Then product "product1" should have following values:
+        | price              | 200             |
+        | ecotax             | 5.5             |
+        | tax rules group    | US-AL Rate (4%) |
+        | on_sale            | false           |
+        | wholesale_price    | 70              |
+#        #todo: rounding issue. #19620
+#        #| unit_price         | 900         |
+        | unity              | bag of ten      |
+        | unit_price_ratio   | 0.222222        |
+
+      Scenario: I update product prices with negative values
+        Given I add product "product2" with following information:
+          | name       | en-US: white hat  |
+          | is_virtual | false             |
+        And I update product "product2" prices with following information:
+          | price           | 50           |
+        And product "product2" has following values:
+          | price           | 50           |
+          | ecotax          | 0            |
+          | wholesale_price | 0            |
+          | unit_price      | 0            |
+        When I update product "product2" prices with following information:
+          | price           | -20          |
+        Then I should get error that product "price" is invalid
+        When I update product "product2" prices with following information:
+          | ecotax          | -2           |
+        Then I should get error that product "ecotax" is invalid
+        When I update product "product2" prices with following information:
+          | wholesale_price | -35          |
+        Then I should get error that product "wholesale price" is invalid
+        When I update product "product2" prices with following information:
+          | unit_price      | -300         |
+        Then I should get error that product "unit price" is invalid
+
+      Scenario: I update product unit price without when product price is 0
+        Given I add product "product3" with following information:
+          | name       | en-US: black hat  |
+          | is_virtual | false             |
+        And product "product3" has following values:
+          | price           | 0            |
+          | unit_price      | 0            |
+        When I update product "product3" prices with following information:
+          | unit_price      | 300          |
+        Then I should get error that product "unit price" is invalid
```

Comment:
It's actually weird to have the same exception for this use case as when you try to set a negative unit price

Maybe this kind of error would deserve a dedicated error code, especially because it's weird to have an invalid error code matching `unit_price` whereas it's also related to `price` field (changing the input for unit_price only will never be enough)

Reply:
removed this exception at all for simplicity. Now if the price is 0 then unit_price is always reset to 0 too.